src/plugins/controls/common/control_group/control_group_persistence.ts

```diff
@@ -57,30 +58,33 @@ export const getDefaultControlGroupPersistableInput = (): PersistableControlGrou
 
 export const persistableControlGroupInputIsEqual = (
   a: PersistableControlGroupInput | undefined,
-  b: PersistableControlGroupInput | undefined
+  b: PersistableControlGroupInput | undefined,
+  compareSelections: boolean = true
 ) => {
-  const defaultInput = getDefaultControlGroupInput();
+  const defaultInput = getDefaultControlGroupPersistableInput();
   const inputA = {
     ...defaultInput,
-    ...pick(a, ['panels', 'chainingSystem', 'controlStyle', 'ignoreParentSettings']),
+    ...pick(a, persistableControlGroupInputKeys),
   };
   const inputB = {
     ...defaultInput,
-    ...pick(b, ['panels', 'chainingSystem', 'controlStyle', 'ignoreParentSettings']),
+    ...pick(b, persistableControlGroupInputKeys),
   };
 
-  if (
-    getPanelsAreEqual(inputA.panels, inputB.panels) &&
-    deepEqual(omit(inputA, 'panels'), omit(inputB, 'panels'))
-  )
-    return true;
-
-  return false;
+  return (
+    Boolean(inputA.showApplySelections) === Boolean(inputB.showApplySelections) &&
```

Comment:
In this case - you are actually correct 🙈 I was influenced from a past issue we had with options list controls where, when doing `deepEqual`, a key being undefined is considered **not equal to** that key being `false`.
Therefore, consider the following steps:

1. Create a brand new dashboard and save it - the `showApplySelections` key **does not exist** in the control group input, but default.
2. Enable `showApplySelection` - unsaved changes will be triggered, as expected.a
3. Disable `showApplySelections` - unsaved changes will **remain**, because `false !== <key does not exist>` when using `deepEqual` 

The reason this **doesn't** happen is because we are spreading `getDefaultControlGroupPersistableInput()` on the input before comparison - that way, `showApplySelections` **always has a value**. So, I can remove this.

Reply:
Removed in 4441d11813106356605d0922e3a00bc684ba496d :+1: Good catch